### PR TITLE
Add more CI targets for cross compilation

### PR DESCRIPTION
**Pull request by carllerche**
This should hopefully catch most errors related to secondary platforms.

### CODE REVIEW

File: ci/azure-cross-compile.yml [hunk 1]
```diff
@@ -26,6 +34,18 @@ jobs:
         vmImage: ubuntu-16.04
         target: x86_64-sun-solaris
 
+      FreeBSD_32:
+        vmImage: ubuntu-16.04
```
Comment (asomers) [inline ci/azure-cross-compile.yml]: Why cross-compile FreeBSD i386 from Linux?  Better to do it from FreeBSD amd64.  That way we can run the tests, too.  See https://github.com/nix-rust/nix/blob/master/.cirrus.yml .

Comment (carllerche) [reply ci/azure-cross-compile.yml]: @asomers either way, this cross compilation doesn't seem to catch the error?

Comment (asomers) [reply ci/azure-cross-compile.yml]: What error are you expecting to catch?

Comment (carllerche) [reply ci/azure-cross-compile.yml]: This fix: https://github.com/tokio-rs/mio/commit/aeaebb0a908f65910ef5cca2863e71734d77d0aa

**Comment by carllerche** [comment]
Ok, I guess running the 32bit freebsd tests is good enough for now... I will rebase against master.

File: ci/azure-cross-compile.yml [hunk 2]
```diff
@@ -26,6 +34,14 @@ jobs:
         vmImage: ubuntu-16.04
         target: x86_64-sun-solaris
 
+      Linux_ARM:
```
Comment (Thomasdezeeuw) [inline ci/azure-cross-compile.yml]: Is the name accurate when using android? I don't what, if any, differences there are still.

File: .cirrus.yml [hunk 3]
```diff
@@ -13,9 +13,11 @@ task:
...
     - cargo build
     - cargo build --no-default-features
-  test_script:
+  amd64_test_script:
     - . $HOME/.cargo/env
     - cargo test
-    - cargo test --no-default-features
+  i386_test_script:
+    - . $HOME/.cargo/env
+    - cargo test --target i686-unknown-freebsd
```
Comment (Thomasdezeeuw) [inline .cirrus.yml]: CI fails, you might need to add `rustup target add i686-unknown-freebsd`.

File: ci/azure-cross-compile.yml (hunk 2 above)
Comment (carllerche) [reply ci/azure-cross-compile.yml]: i did not put much thought into the name...

**Comment by Thomasdezeeuw** [comment]
Done in #1057.

**Pull request closed without merging**